File: Library/Models/Book.cs
using System.ComponentModel.DataAnnotations;
using System.Text.Json.Serialization;

namespace Library.Models
{
    public class Book
    {
        [Key]
        public int BookId { get; set; }

        [MaxLength(50)]
        public string Title { get; set; }

        public string  Author { get; set; }

        [JsonConverter(typeof(JsonStringEnumConverter))]
        public Genre BookGenre { get; set; }

        [DisplayFormat(DataFormatString ="{0:yyyy}")]
        public DateTime RealeseYear { get; set; }

        public enum Genre
        {
            Deckare,
            Biografi,
            Romantik,
            Fantasy,
            Scifi,
            Självbiografi
        }
    }
}

File: LibraryFEAngular/library-angular/src/app/Models/Book.ts
export interface Book {
    bookId: string; // Book ID
    title: string; // Book Title
    author: string; // Book Author
    bookGenre: string; // Book Genre        
    realeseYear: string; // Book Release
    } // Book Genre        
    export enum BookGenre {
        Deckare = 'Deckare',
        Biografi = 'Biografi',
        Romantik = 'Romantik',
        Fantasy = 'Fantasy',
        ScienceFiction = 'ScienceFiction',
        Självbiografi = 'Självbiografi',
    }